core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts

```diff
@@ -173,6 +173,31 @@ export class BaseUser {
     inputUploadHandle.uploadFile(fileToUpload);
   }
 
+  /**
+   * This function validates whether an anchor tag is correctly linked
+   * to external PDFs or not. Use this particularly when interacting with
+   * buttons associated with external PDF links, because Puppeteer in
+   * headless mode, does not natively support opening of external PDFs.
+   */
+  async openExternalPdfLink(
+    selector: string,
+    expectedUrl: string
+  ): Promise<void> {
+    await this.page.waitForSelector(selector);
+    const href = await this.page.$eval(selector, element =>
+      element.getAttribute('href')
+    );
+    if (href === null) {
+      throw new Error(`The ${selector} is not an anchor tag!`);
```

Comment:
Technically to give this error message you'll need to check that it's not an "a" tag. But what you're checking here is that the selector has an href attribute.

Have your error message match the code so that it's not misleading.